File: CS50x/sentimental-readability/readability.py
import cs50

text = cs50.get_string("Text: ")


def count_letters(text):
    letter = 0
    for char in text:
        if char.isalpha():
            letter += 1
    return letter


def count_words(text):
    word = 1
    for i in range(1, len(text)):
        if text[i].isspace() and not text[i - 1].isspace():
            word += 1
    return word


def count_sentences(text):
    sentence = 0
    for char in text:
        if char in ('.', '!', '?'):
            sentence += 1
    return sentence


letters = count_letters(text)
words = count_words(text)
sentences = count_sentences(text)


L = letters / words * 100
S = sentences / words * 100
index = 0.0588 * L - 0.296 * S - 15.8
grade = round(index)


if grade < 1:
    print("Before Grade 1")

elif grade > 16:
    print("Grade 16+")

else:
    print(f"Grade {grade}")

File: CS50x/readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    // Prompt the user for some text
    string text = get_string("Text: ");

    // Count the number of letters, words, and sentences in the text
    int letters = count_letters(text);
    int words = count_words(text);
    int sentences = count_sentences(text);

    // Compute the Coleman-Liau index
    float L = (float) letters / words * 100;
    float S = (float) sentences / words * 100;
    float index = 0.0588 * L - 0.296 * S - 15.8;
    int grade = round(index);

    // Print the grade level
    if (grade < 1)
    {
        printf("Before Grade 1\n");
    }

    else if (grade > 16)
    {
        printf("Grade 16+\n");
    }

    else
    {
        printf("Grade %i\n", grade);
    }
}

int count_letters(string text)
{
    // Return the number of letters in text
    int letter = 0;
    for (int i = 0; text[i] != '\0'; i++)
    {
        if (isalpha(text[i]))
        {
            letter++;
        }
    }
    return letter;
}

int count_words(string text)
{
    // Return the number of words in text
    int word = 1;
    for (int i = 0; text[i] != '\0'; i++)
    {
        if ((isblank(text[i]) && !isblank(text[i - 1])))
        {
            word++;
        }
    }
    return word;
}

int count_sentences(string text)
{
    // Return the number of sentences in text
    int sentence = 0;
    for (int i = 0; text[i] != '\0'; i++)
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentence++;
        }
    }
    return sentence;
}
